jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
```

Comment:
`map` is indeed correct here.

`andThen` is, to be honest, a wtf.  Usually I know that function as `bind`, never seen it named `andThen` before and that name does not make sense to me...  `bind` though is not much more descriptive really, but `bind` is the common name for it.